tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,375 @@
+        'reposet': 'Red Hat Enterprise Linux 7 Server (RPMs)',
+        'product': 'Red Hat Enterprise Linux Server',
+    },
+    'rhel7_server_extras': {
+        'id': 'rhel-7-server-extras-rpms',
+        'name': 'Red Hat Enterprise Linux 7 Server - Extras RPMs x86_64',
+        'releasever': '7',
+        'reposet': 'Red Hat Enterprise Linux 7 Server - Extras (RPMs)',
+        'product': 'Red Hat Enterprise Linux Server',
+    },
+    'rhel8_bos': {
+        'id': 'rhel-8-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - BaseOS RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
+        'reposet': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel8_aps': {
+        'id': 'rhel-8-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - AppStream RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
+        'reposet': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)',
+    },
+    'rhel9_bos': {
+        'id': 'rhel-9-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - BaseOS RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel9_aps': {
+        'id': 'rhel-9-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - AppStream RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)',
+    },
+}
+
+
+@pytest.fixture
+def function_leapp_cv(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Content View')
+    return module_target_sat.api.ContentView(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture(scope='module')
+def module_leapp_lce(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Lifecycle Environment')
+    return module_target_sat.api.LifecycleEnvironment(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture
+def function_leapp_ak(
+    module_target_sat, function_leapp_cv, module_leapp_lce, module_sca_manifest_org, upgrade_path
+):
+    logger.info(
+        'Creating Leapp Activation Key for RHEL %s -> %s',
+        upgrade_path['source_version'],
+        upgrade_path['target_version'],
+    )
+    # need to publish and promote cv on lce to get content available on client host
+    function_leapp_cv.publish()
+    cvv = function_leapp_cv.read().version[0]
+    cvv.promote(data={'environment_ids': module_leapp_lce.id, 'force': True})
+    function_leapp_cv = function_leapp_cv.read()
+    ak = module_target_sat.api.ActivationKey(
+        content_view=function_leapp_cv,
+        environment=module_leapp_lce,
+        organization=module_sca_manifest_org,
+    ).create()
+    # In case of 'rhel7_server_extras' repos, need to enabled (overridden) repository
+    # to get content available on client host
```

Comment:
Can we have a check for this in AK, whether any repo is left disabled then it can be enabled by override?

Reply:
- This is special case, `rhel7_server_extras` repo is just added to CV but not enabled on the host > AK > content host

Reply:
@vijaysawant 
1. Your reply does not answer Gaurav's question.
2. What did the Phoenix team say about this behavior? Is it the expected behavior of Satellite that this repo is not enabled?

Reply:
Yes, @vijaysawant extras isn't default enabled i know, this is default behaviour in 6.14, so it'd be better to have a check for this in AK, whether any repo is left disabled then it can be enabled by override?

Reply:
I have changed my mind on this after a deeper look.
Maybe the following code could be moved to the `function_leapp_cv` fixture:

```python
    function_leapp_cv.repository = all_repos  # all_repos would become leapp_repos, see below
    function_leapp_cv = function_leapp_cv.update(['repository'])
    logger.info('Repos to be added to the AK: %s', all_repos)
    # Publish, promote content view to lce
    logger.info(f'Publish, promote cv - {function_leapp_cv.name} to lce - {module_leapp_lce.name}')
    function_leapp_cv.publish()
    cvv = function_leapp_cv.read().version[0]
    cvv.promote(data={'environment_ids': module_leapp_lce.id, 'force': True})
    function_leapp_cv = function_leapp_cv.read()
```

In this case,  the `leapp_sat_content` fixture would need to return `all_repos` and it needs to be a dependency of the `function_leapp_cv` fixture.
Also, it would make sense to rename the `leapp_sat_content` to something like `leapp_repos` as it would only manage repositories.

@vijaysawant What do you think about these potential changes? At least IMO this should be easy to implement.

Reply:
ah I see, just checked, I was under impression that we're adding both scenarios here, my bad, sorry, will resolve this comment then, and will nitpick again with this later :)